pkg/release/hook.go

```diff
@@ -60,9 +71,19 @@ type Hook struct {
 	// Events are the events that this hook fires on.
 	Events []HookEvent `json:"events,omitempty"`
 	// LastRun indicates the date/time this was last run.
-	LastRun time.Time `json:"last_run,omitempty"`
+	LastRun HookExecution `json:"last_run,omitempty"`
 	// Weight indicates the sort order for execution among similar Hook type
 	Weight int `json:"weight,omitempty"`
 	// DeletePolicies are the policies that indicate when to delete the hook
 	DeletePolicies []HookDeletePolicy `json:"delete_policies,omitempty"`
 }
+
+// A HookExecution records the result for the last execution of a hook for a given release.
+type HookExecution struct {
+	// StartedAt indicates the date/time this hook was started
+	StartedAt time.Time `json:"started_at,omitempty"`
+	// CompletedAt indicates the date/time this hook was completed
+	CompletedAt time.Time `json:"completed_at,omitempty"`
+	// Successful indicates whether the hook completed successfully
+	Successful bool `json:"successful"`
```

Comment:
Should this be a hook phase instead, similar to pod phases? If I'm checking on the status of a hook execution event, how can I infer that it's in an unknown state instead of a pending state?

Perhaps this is related to my earlier comment on `waitForPodSuccess`.

There is also a TestRun struct in `pkg/release` that includes an `Info` and a `Status` object. It appears similar to this object's intended use. Status can be one of "success", "failed", "running" or "unknown". Perhaps that struct and this one could be consolidated.

Reply:
Thanks, I missed `TestRun` earlier. Refactored in fada0c92f8ed47964dced8a639f841a8afaff931.